src/uwtools/config/formats/nml.py

```diff
@@ -52,31 +44,26 @@ def dump(self, path: OptionalPath) -> None:
 
         :param path: Path to dump config to.
         """
-        config_check_depths_dump(config_obj=self, target_format=FORMAT.nml)
-
-        self.dump_dict(self.data, path)
+        self.dump_dict(cfg=self.data, path=path)
 
     @staticmethod
-    def dump_dict(cfg: dict, path: OptionalPath = None) -> None:
+    def dump_dict(cfg: Union[dict, Namelist], path: OptionalPath = None) -> None:
```

Comment:
This method needs to support the case where an existing `NMLConfig` object is being dumped, in which case its type-`Namelist` `data` attribute will be passed in as `cfg`; and the case where an arbitrary `dict` object is supplied by a caller.